python/ray/experimental/data/dataset.py

```diff
@@ -805,8 +805,31 @@ def iter_batches(self,
         Returns:
             A list of iterators over record batches.
         """
-
-        raise NotImplementedError  # P1
+        blocks = list(iter(self._blocks))
+        for curr_block_idx in range(len(blocks)):
+            # Prefetch blocks.
+            ray.wait(
+                blocks[curr_block_idx:curr_block_idx + prefetch_blocks],
+                num_returns=1,
+                fetch_local=True)
+            # Get the current block.
+            block = ray.get(blocks[curr_block_idx])
+            total_rows = block.num_rows()
```

Comment:
Maybe leave a TODO to combine batches across blocks?

Reply:
I've implemented similar logic for the shuffling data loader, I'll port that logic in this PR.